pac4j-core/src/main/java/org/pac4j/core/profile/UserProfile.java

```diff
@@ -154,12 +156,18 @@ public String getTypedId() {
     }
 
     /**
-     * Get attributes as immutable map.
+     * Get all attributes as immutable map.
      * 
      * @return the immutable attributes
      */
     public Map<String, Object> getAttributes() {
-        return Collections.unmodifiableMap(this.attributes);
+        final Map<String, Object> newAttributes = new HashMap<>();
+        for (Map.Entry<String, Object> entries : this.attributes.entrySet()) {
+            final String key = entries.getKey();
+            final Object value = getAttribute(key);
+            newAttributes.put(key, value);
+        }
+        return Collections.unmodifiableMap(newAttributes);
```

Comment:
Of course, while I can see the usage of method be problematic in general, I can't quite foresee a practical issue so this may just be moot for now, but I'd prefer to play safe specially as this PR handles CAS-related changed, and CAS certainly prefers things to be serializable friendly.